pyomo/core/base/indexed_component.py

```diff
@@ -666,6 +666,27 @@ def _processUnhashableIndex(self, idx):
                 fixed[i - len(idx)] = val
 
         if sliced or ellipsis is not None:
+            if self.dim() is None:
+                # Assume that the right thing to do here is return
+                # an IndexedComponent_slice
+                pass
+            elif self.dim() == 0 and idx == (slice(None),):
+                # If dim == 0 and idx is slice(None), the component was
+                # a scalar passed a single slice. Since scalar components
+                # can be accessed with a "1-dimensional" index of None,
+                # this behavior is allowed.
+                pass
+            elif ((ellipsis is None and len(idx) != self.dim()) or 
+                # If there is no ellipse and the index doesn't match the 
+                # component's dimension, raise an error.
+                # NOTE: Should this do something different for unflattened 
+                # indices?
+                (len(fixed) + len(sliced) > self.dim())):
+                # If an ellipsis is present and the index exceeds the dimension
+                # of the component, raise an error.
```

Comment:
Bit of devil's advocate: is there a use case of slicing where you'd slice a slice?

I personally cannot think of a good one right now.

Reply:
I'm not sure. To achieve the functionality you'd presumably want by slicing a slice, I would slice a reference-to-slice. What is the implication of being able to slice a slice? And to be clear, you're talking about trying to run code like this?
```
m.v[:,2,:][:,'a'].fix(0)
```